src/server/api/routers/bot-source.ts

```diff
@@ -51,6 +52,33 @@ function deleteByIdHandler() {
     })
 }
 
+function toggleVisibility() {
+  return protectedProcedure
+    .input(
+      z.object({
+        botSourceId: z.string(),
+      }),
+    )
+    .mutation(async ({ input }) => {
+      const bs = await db
+        .select({
+          id: schema.botSources.id,
+          visible: schema.botSources.visible,
+        })
+        .from(schema.botSources)
+        .where(eq(schema.botSources.id, input.botSourceId))
```

Comment:
Let's also select where created_by == ctx.session.user.id
to ensure bot's authentication.